### Assignment 12 KNN 
Implement a KNN model to classify the glass in to categorie

In [1]:
import pandas as pd
import numpy as np
from sklearn.model_selection import KFold,cross_val_score
from sklearn.neighbors import KNeighborsClassifier

In [2]:
from sklearn.model_selection import GridSearchCV

In [3]:
glass=pd.read_csv("glass.csv")
glass.head()

RI     Na    Mg    Al     Si     K    Ca   Ba   Fe  Type
0  1.52101  13.64  4.49  1.10  71.78  0.06  8.75  0.0  0.0     1
1  1.51761  13.89  3.60  1.36  72.73  0.48  7.83  0.0  0.0     1
2  1.51618  13.53  3.55  1.54  72.99  0.39  7.78  0.0  0.0     1
3  1.51766  13.21  3.69  1.29  72.61  0.57  8.22  0.0  0.0     1
4  1.51742  13.27  3.62  1.24  73.08  0.55  8.07  0.0  0.0     1

In [4]:
glass.shape

(214, 10)

In [5]:
x=glass.iloc[:,0:9]
y=glass.iloc[:,9]


In [6]:
kfold=KFold(n_splits=200)
model_1=KNeighborsClassifier(n_neighbors=1)

In [7]:
result=cross_val_score(model_1,x,y,cv=kfold)
result.mean()

0.735

In [10]:
from sklearn.model_selection import GridSearchCV

#### Grid search cv

In [13]:
model = KNeighborsClassifier()
param_grid=dict(n_neighbors=np.array(range(1,10)))

gsv =GridSearchCV(model,param_grid=param_grid)
gsv.fit(x,y)

GridSearchCV(estimator=KNeighborsClassifier(),
             param_grid={'n_neighbors': array([1, 2, 3, 4, 5, 6, 7, 8, 9])})

In [16]:
gsv.best_params_

{'n_neighbors': 1}

In [15]:
gsv.best_score_

0.640531561461794

#### Conclusion -

#### Our Accuracy is 64 % when we take 1 neighboring classifier